csFastFloat/Utils/Utils.cs

```diff
@@ -66,7 +73,52 @@ internal static bool is_integer(char c, out uint cMinus0)
       cMinus0 = cc;
       return res;
     }
-    
+
+
+
+#if HAS_INTRINSICS
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    unsafe internal static bool eval_parse_eight_digits_simd(char* start, char* end, out uint value)
+    {
+
+      value = 0;
+
+      Vector128<short> raw = Sse41.LoadDquVector128((short*)start);
+      Vector128<short> ascii0 = Vector128.Create((short)47);
+      Vector128<short> after_ascii9 = Vector128.Create((short)58);
+
+      var a = Sse41.CompareGreaterThan(raw, ascii0);
+      var b = Sse41.CompareLessThan(raw, after_ascii9);
+      var c = Sse41.Subtract(a, b);
+
+      if (!Sse41.TestZ(c, c))
+        return false;
+
+      // Credit : @aepot
+      // https://stackoverflow.com/questions/66371621/hardware-simd-parsing-in-c-sharp-performance-improvement/66430672
+      Vector128<short> mask0 = Vector128.Create((short)48); // adapter à la longueur
+      raw = Sse41.SubtractSaturate(raw, mask0);
+      Vector128<short> mul0 = Vector128.Create(10, 1, 10, 1, 10, 1, 10, 1);
+      Vector128<int> res = Sse41.MultiplyAddAdjacent(raw.AsInt16(), mul0);
+      Vector128<int> mul1 = Vector128.Create(1000000, 10000, 100, 1);
+      res = Sse41.MultiplyLow(res, mul1);
```

Comment:
AFAIK, `pmulld` is slightly more expensive than the (my answer on stackoverflow) alternative, though I haven't measured it.

Reply:
@aqrit Can you point at your version. I am sure one can come up with something slightly more efficient, but it would be useful to have your code to compare.